src/s3_plugin.js

```diff
@@ -312,4 +322,113 @@ module.exports = class S3Plugin {
       }
     })
   }
+
+  setCloudfrontIndex(clientConfig, cloudfrontInvalidateOptions, indexOptions) {
+    return new Promise(function(resolve, reject) {
+      // Setup Cloudfront
+      var cloudfront = new aws.CloudFront()
+      cloudfront.config.update({
+        accessKeyId: clientConfig.s3Options.accessKeyId,
+        secretAccessKey: clientConfig.s3Options.secretAccessKey,
+      });
+
+      // Get the existing distribution
+      cloudfront.getDistribution({
+        Id: cloudfrontInvalidateOptions.DistributionId
+      }, (err, data) => {
+        if (err) {
+          reject(err)
+        } else {
+          if (data.DistributionConfig.DefaultRootObject === indexOptions.IndexDocument) {
+            resolve()
+          }
+
+          // Update the distribution with the new default root object
+          data.DistributionConfig.DefaultRootObject = indexOptions.IndexDocument;
+
+          cloudfront.updateDistribution({
+            IfMatch: data.ETag,
+            Id: cloudfrontInvalidateOptions.DistributionId,
+            DistributionConfig: data.DistributionConfig
+          }, function(err, data) {
+            if (err) {
+              reject(err)
+            } else {
+              resolve()
+            }
+          });
+        }
+      });
+    });
+  }
+
+  setS3Index(clientConfig, uploadOptions, indexOptions) {
+    return new Promise(function(resolve, reject) {
+      var s3Client = new aws.S3({
```

Comment:
Creating a client is not necessary. It's already created [here](https://github.com/happylinks/s3-plugin-webpack/blob/7fa8daa30d48ea3e7f7dcbee62e5d83bd4e1705e/src/s3_plugin.js#L78)